### DISCUSSION TITLE
Is there a URL to download the latest release of GHA runner?

### DISCUSSION THREAD

**Discussion by vitalyk-multinarity** [community/community#54078] [Actions] [answered]
### Select Topic Area

Question

### Body

For launching self-hosted runner we should download GHA runner tarball from https://github.com/actions/runner/releases. Is there a way to download the latest v2 release instead of updating download URL every few months?

**Comment by hrishikesh-kocharekar-dev** [comment, marked as answer]
Yes, there is a URL that always points to the latest release of the self-hosted GitHub Actions runner. You can use the following URL to download the latest release tarball: https://github.com/actions/runner/releases/latest/download/actions-runner-linux-x64-<version>.tar.gz

Replace `<version>` with the specific version of the runner you want to download (e.g., `v2.278.0)`, and the URL will always download the latest release of the runner with that version number.

Keep in mind that using the `latest` tag may not always be suitable for production environments, as new releases can potentially introduce breaking changes or bugs that may affect your workflows. So it's a good practice to test the latest release in a staging environment before upgrading your production environment.

**Reply by vitalyk-multinarity** [reply]
Thank you! Agree that using the latest may be problematic, so ideally I'd like to get the latest v2 build.

**Comment by kiloforce** [comment]
Did I misunderstand the directions?  URL above doesn't seem to download latest.  The directory resolved the latest version correctly, but the filename did not...

```
$ wget https://github.com/actions/runner/releases/latest/download/actions-runner-linux-x64-.tar.gz
--2023-10-02 11:17:15--  https://github.com/actions/runner/releases/latest/download/actions-runner-linux-x64-.tar.gz
Resolving github.com (github.com)... 140.82.112.4
Connecting to github.com (github.com)|140.82.112.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github.com/actions/runner/releases/download/v2.309.0/actions-runner-linux-x64-.tar.g [following]
--2023-10-02 11:17:16--  https://github.com/actions/runner/releases/download/v2.309.0/actions-runner-linux-x64-.tar.gz
Reusing existing connection to github.com:443.
HTTP request sent, awaiting response... 404 Not Found
2023-10-02 11:17:16 ERROR 404: Not Found.
```

**Comment by kiloforce** [comment]
This little script worked for me, based on the latest directory redirecting:

```
LATEST=`curl -s -i https://github.com/actions/runner/releases/latest | grep location:`
LATEST=`echo $LATEST | sed 's#.*tag/v##'`
LATEST=`echo $LATEST | sed 's/\r//'`
curl -L -s "https://github.com/actions/runner/releases/download/v${LATEST}/actions-runner-linux-x64-${LATEST}.tar.gz" -o actions-runner-linux-x64-${LATEST}.tar.gz
```

**Reply by audunmg** [reply]
That's a nice script.

You can compact it by running it all through a single sed:
```bash
LATEST=`curl -s -I https://github.com/actions/runner/releases/latest | sed -n '/^location:/{ s#.*tag/v##; s/\r//g; p; }' ` 
curl -L -s "https://github.com/actions/runner/releases/download/v${LATEST}/actions-runner-linux-x64-${LATEST}.tar.gz" -o actions-runner-linux-x64-${LATEST}.tar.gz
```

* `sed -n` makes sed not print every line by default
* `/^location:/` will match the line starting with location: and execute the statement in the `{ }` block 
* Then I just copypasted your sed expressions after each other in the block, end each with semicolon `;` and lastly, printing the line with `p;` 

Sed is nice :)

**Comment by shekhfaisal2110** [comment]


**Reply by audunmg** [reply]
Option 1 results in 404 not found. Did you test it?
